### PR TITLE
#6667 feat: Gallery thumbnail view (WIP)

### PR DESCRIPTION
**Pull request by creido-welly**
- Lead image logic and styling
- Abstracted gallery images out as GalleryMedia component
- Abstracted img elements from Gallery and Image as ImageElement to allow for common setting of responsive attributes

### CODE REVIEW

File: src/components/Gallery/_gallery.scss [hunk 1]
```diff
@@ -36,21 +39,38 @@
...
 }
 
 .cc-gallery__media-content {
   @extend %image-responsive;
 }
+
+.break {
+  flex-basis: 100%;
+  height: 0;
+}
```
Comment (chriddell) [inline src/components/Gallery/_gallery.scss:60]: What does this do?

Comment (creido-welly) [reply src/components/Gallery/_gallery.scss:60]: aha! magic (ssh)

alright then, it's a means of forcing flexbox items on to the next row. In this instance I'm using it to ensure the lead image (if there is one) is the only item on its row. Not the most helpfully named class right now I'll admit - how about `u-flexgrid-break`?

Comment (chriddell) [reply src/components/Gallery/_gallery.scss:60]: ah okay - would `flex-wrap: wrap` on the parent work, instead?

Comment (creido-welly) [reply src/components/Gallery/_gallery.scss:60]: sadly not - as discussed this does appear to be the neatest solution to the problem

Comment (chriddell) [reply src/components/Gallery/_gallery.scss:60]: agreed

**Review by chriddell — APPROVED** [review]

**Pull request merged**